oak_runtime/src/node/logger.rs

```diff
@@ -47,51 +47,66 @@ impl super::Node for LogNode {
+                            // whether the Runtime has been built with the `oak_debug` feature
+                            // enabled (and thus whether log! is connected up to anything or not).
+                            // So send to stderr.
+                            let now: chrono::DateTime<chrono::Utc> = chrono::Utc::now();
+                            let timestamp = now.format("%Y-%m-%dT%H:%M:%SZ");
+                            eprintln!(
+                                "{{{} {} {}:{}}} {}",
+                                timestamp,
+                                level_to_string(msg.level),
+                                msg.file,
+                                msg.line,
+                                msg.message
+                            );
+                        }
+                        Err(error) => {
+                            error!("{} Could not parse LogMessage: {}", self.node_name, error)
+                        }
+                    },
+                    Ok(None) => {
+                        // No more log messages to read, go back to waiting.
+                        break 'read;
+                    }
+                    Err(OakStatus::ErrChannelClosed) => {
+                        info!("{} channel closed by remote", self.node_name);
+                        break 'wait;
+                    }
+                    Err(status) => {
+                        error!("{} Failed channel read: {:?}", self.node_name, status);
+                        break 'wait;
                     }
-                },
-                Ok(None) => {}
-                Err(OakStatus::ErrChannelClosed) => {
-                    break;
-                }
-                Err(status) => {
-                    error!("{} Failed channel read: {:?}", self.node_name, status);
-                    break;
                 }
             }
+
+            if result == Err(OakStatus::ErrTerminated) {
+                break 'wait;
+            }
         }
         info!("{} logger execution complete", self.node_name);
         let _ = runtime.channel_close(handle);
     }
 }
 
-/// Translate a log level as encoded in a `LogMessage` into the corresponding
-/// enum value from the [`log` crate](https://crates.io/crates/log).
-fn to_level(level: i32) -> log::Level {
+/// Translate a log level as encoded in a `LogMessage` into a string
+fn level_to_string(level: i32) -> &'static str {
     match Level::from_i32(level).unwrap_or(Level::UnknownLevel) {
-        Level::UnknownLevel => log::Level::Error,
-        Level::Error => log::Level::Error,
-        Level::Warn => log::Level::Warn,
-        Level::Info => log::Level::Info,
-        Level::Debugging => log::Level::Debug,
-        Level::Trace => log::Level::Trace,
+        Level::UnknownLevel => " N/A ",
+        Level::Error => "ERROR",
+        Level::Warn => "WARN ",
+        Level::Info => "INFO ",
+        Level::Debugging => "DEBUG",
+        Level::Trace => "TRACE",
```

Comment:
It looks like the intention is to pad all names to N character, so perhaps specify that in the comment (and add a trivial test for it if you feel like it).

Reply:
Added a comment (it's purely cosmetic so I don't think it's worth testing for).

Reply:
Yeah it was more a case of test as machine-checked documentation :) anyways by all means feel free to ignore.